From 46d7b017b2c89617c2c0060b7bb9d266ebf75887 Mon Sep 17 00:00:00 2001
From: est31 <MTest31@outlook.com>
Date: Wed, 10 Aug 2016 09:30:39 +0200
Subject: [PATCH] Allow XDG_DATA_DIR as override for path_user

Needed by flatpak
---
 src/porting.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/porting.cpp b/src/porting.cpp
index 98b85b7..e4cab10 100644
--- a/src/porting.cpp
+++ b/src/porting.cpp
@@ -552,6 +552,12 @@ void initializePaths()
 	}
 	// Migrate cache folder to new location if possible
 	migrateCachePath();
+
+	// If XDG_DATA_HOME is set, override path_user
+	const char *path_user_data = getenv("XDG_DATA_HOME");
+	if (path_user_data) {
+		path_user = std::string(path_user_data);
+	}
 #endif
 
 	infostream << "Detected share path: " << path_share << std::endl;
